velox/functions/sparksql/IntervalFunctions.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <iostream>
+#include "velox/functions/sparksql/DateTimeFunctions.h"
+
+namespace facebook::velox::functions::sparksql {
+
+inline int64_t safeMul(int64_t a, int64_t b, const char* context) {
+  int64_t result;
+  VELOX_USER_CHECK(
+      !__builtin_mul_overflow(a, b, &result),
+      "Overflow during multiplication in {}: {} * {}",
+      context,
+      a,
+      b);
+  return result;
+}
+
+inline int64_t safeAdd(int64_t a, int64_t b, const char* context) {
+  int64_t result;
+  VELOX_USER_CHECK(
+      !__builtin_add_overflow(a, b, &result),
+      "Overflow during addition in {}: {} + {}",
+      context,
+      a,
+      b);
+  return result;
+}
+
+template <typename T>
+struct MakeDTIntervalFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<IntervalDayTime>& result,
+      int32_t days = 0,
+      int32_t hours = 0,
+      int32_t minutes = 0,
+      double secondsMicros = 0) {
```

Comment:
I assume the fourth argument is the unscaled value of short decimal type with scale 6, and its C++ type is int64_t.